### DISCUSSION TITLE
"node-exporter-bt42h"  is  "CrashLoopBackOff"

### DISCUSSION THREAD

**Discussion by evan-zxx** [prometheus-operator/kube-prometheus#601] [General]
**What happened?**
my operation:

0: branch to the git version "/origin/release-0.3"

1: run the scripts ./scripts/monitoring-deploy.sh

2: kubectl get pod -n monitoring
NAME                                   READY   STATUS             RESTARTS   AGE
alertmanager-main-0                    2/2     Running            0          68m
alertmanager-main-1                    2/2     Running            0          68m
alertmanager-main-2                    0/2     Pending            0          65m
grafana-5db74b88f4-phmmt               1/1     Running            0          169m
kube-state-metrics-54f98c4687-pfsx7    3/3     Running            0          169m
node-exporter-bt42h                    1/2     CrashLoopBackOff   37         169m
prometheus-adapter-8667948d79-kfs8c    1/1     Running            0          169m
prometheus-k8s-0                       0/3     Pending            0          67m
prometheus-k8s-1                       0/3     Pending            0          65m
prometheus-operator-5f759fd859-vbwk7   1/1     Running            0          68m

3: kubectl describe pod node-exporter-bt42h -n monitoring
Warning  BackOff  2m (x694 over 156m)  kubelet, docker-desktop  Back-off restarting failed container
State:          Waiting
      Reason:       CrashLoopBackOff
    Last State:     Terminated
      Reason:       ContainerCannotRun
      Message:      path / is mounted on / but it is not a shared or slave mount
      Exit Code:    128

4:  kubectl logs node-exporter-bt42h -c node-exporter -n monitoring
failed to open log file "/var/log/pods/monitoring_node-exporter-bt42h_a4e62f25-bf3d-11ea-a43c-025000000001/node-exporter/35.log": open /var/log/pods/monitoring_node-exporter-bt42h_a4e62f25-bf3d-11ea-a43c-025000000001/node-exporter/35.log: no such file or directory

5: so..how..

**Did you expect to see some different?**

**How to reproduce it (as minimally and precisely as possible)**:

**Environment**
on macOS - docker Desktop 

Client Version: version.Info{Major:"1", Minor:"14", GitVersion:"v1.14.3", GitCommit:"5e53fd6bc17c0dec8434817e69b04a25d8ae0ff0", GitTreeState:"clean", BuildDate:"2019-06-06T01:44:30Z", GoVersion:"go1.12.5", Compiler:"gc", Platform:"darwin/amd64"}

* Prometheus Operator version:
Image:      quay.io/coreos/prometheus-operator:v0.34.0

* Kubernetes version information:
Kubernetes: v1.14.3

* Kubernetes cluster kind:
only one master / for testing

* Manifests:
release-0.3 version
```
insert manifests relevant to the issue
```

* Prometheus Operator Logs:

```
Insert Prometheus Operator logs relevant to the issue here
```

* Prometheus Logs:

```
Insert Prometheus logs relevant to the issue here
```

**Anything else we need to know?**:

**Comment by paulfantom** [comment]
You have a problem with your k8s setup overall. 

```
Reason: CrashLoopBackOff
Last State: Terminated
Reason: ContainerCannotRun
Message: path / is mounted on / but it is not a shared or slave mount
Exit Code: 128
```
This suggests a problem with the mounted filesystem. More on that issue in https://github.com/kubernetes/kubernetes/issues/61058

**Comment by sobvan** [comment]
I have similar issues, and it worked a few month ago with the same setup.
Are you on an M1 chip based Mac @evan-zxx?

**Comment by AlaaHamoudah** [comment]
There is a workaround to remove the mountPropagation to use the default one through creating a patch for promethous exporter damonset, here is the [solution](https://github.com/prometheus-community/helm-charts/issues/467)

**Comment by paulfantom** [comment]
For users who find it later. This is the same issue as described in https://github.com/prometheus-operator/kube-prometheus/discussions/790
